### PR TITLE
Issue #214481: Add option to filter code from copilot chat text to speech

### PR DESCRIPTION
**Pull request by burkus**
This is an implementation of a requested feature issued here: #214481 

I've added a new option to the `accessibility.voice.*` options, called 'FilterCode'.
It is enabled by default. 
When on, it filters out code blocks delimited by triple backticks "```" so that text to speech doesn't sound spammy/annoying when it contains large amounts of code.

I was hoping to do this as part of [Hacktoberfest 
](https://hacktoberfest.com/)

### CODE REVIEW

**Comment by burkus** [comment]
@bpasero thanks for your help!

**Comment by burkus** [comment]
@microsoft-github-policy-service agree

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts [hunk 1]
```diff
@@ -837,10 +840,51 @@ class ChatSynthesizerSessions {
...
+		} else if (ignoreCodeBlocks && !isWithinCodeBlock) {
+			const [filteredText, terminated] = this.withoutCodeBlocks(text);
+			text = filteredText;
+			codeBlockTerminates = terminated;
+		} else if (ignoreCodeBlocks && isWithinCodeBlock && firstDelimiterIndex >= 0) {
+			text = text.substring(firstDelimiterIndex + 1);
+			const [filteredText, terminated] = this.withoutCodeBlocks(text);
+			text = filteredText;
+			codeBlockTerminates = terminated;
+		}
```
Comment (burkus) [inline src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts:876]: @bpasero hope I didn't overcomplicate it

**Comment by bpasero** [comment]
Going with https://github.com/microsoft/vscode/pull/235697, thanks though, much appreciated!

**Pull request closed without merging**